# Project 1: Driving Licenses, Traffic Accidents and Casualties Analysis

## Problem Statment

Craft your problem statement here using the prompt from the README and other sources you find during your investigation.

## Executive Summary
Write an executive summary that summarizes the problem and your key findings.

### Contents:
- [Datasets Description](#Datasets-Description)
- [Data Import & Cleaning](#Data-Import-and-Cleaning)
- [Exploratory Data Analysis](#Exploratory-Data-Analysis)
- [Data Visualization](#Visualize-the-data)
- [Descriptive and Inferential Statistics](#Descriptive-and-Inferential-Statistics)
- [Outside Research](#Outside-Research)
- [Conclusions and Recommendations](#Conclusions-and-Recommendations)

**If you combine your problem statement, executive summary, data dictionary, and conclusions/recommendations, you have an amazing README.md file that quickly aligns your audience to the contents of your project.** Don't forget to cite your data sources!

*All libraries used should be added here, including any Jupyter magic commands*

In [1]:
#Imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

## Datasets Description

[Driving Licenses](https://datasource.kapsarc.org/explore/dataset/saudi-arabia-driving-licenses-issued-in-the-kingdom-2004-2008/information/?disjunctive.administritive_area&sort=time_period&location=5,24.37495,45.08024&basemap=jawg.streets)
This dataset contains Saudi Arabia Driving Licenses Issued By Administrative Area for 1993 - 2016. Data from General Authority for Statistics . Follow datasource.kapsarc.org for timely data to advance energy economics research.

[Traffic Accidents and Casualties](https://datasource.kapsarc.org/explore/dataset/saudi-arabia-traffic-accidents-and-casualties-injured-dead-2008/export/?disjunctive.region&disjunctive.indicator&sort=time_period)
This dataset contains Saudi Arabia Traffic Accidents and Casualties by Region for 2016. Data from General Authority for Statistics. Follow datasource.kapsarc.org for timely data to advance energy economics research.


## Data Import and Cleaning

#### 1. Read In Driving Licenses & Traffic Accidents Data
Read in the `saudi-arabia-traffic-accidents-2008.csv` and `saudi-arabia-driving-licenses-2004-2008.csv` files and assign them to appropriately named pandas dataframes.

In [2]:
# Read in datasets from csv to pandas dataframes, load only useful columns
acc_columns = ['Year', 'Region', 'Indicator', 'Value', 'geo_point_2d']
lic_columns = ['Year', 'Administritive Area', 'Driving Liceses', 'geo_point_2d']

accidents = pd.read_csv('../data/saudi-arabia-traffic-accidents-2008.csv', sep=';', usecols=acc_columns)
licenses = pd.read_csv('../data/saudi-arabia-driving-licenses-2004-2008.csv', sep=';', usecols=lic_columns)

#### 2. Display data

Print the first 10 rows of each dataframe to your jupyter notebook

In [3]:
# Check dataframes using .head() & .tail() and .shape
accidents.head(10)

Year            Region                    Indicator   Value  \
0  (2016)            Makkah  No. of Casualties - Injured  12383n   
1  (2016)    Eastern Region     No. of Casualties - Dead   1113n   
2  (2016)            Tabouk     No. of Casualties - Dead    434n   
3  (2016)              Hail     No. of Casualties - Dead    299n   
4  (2016)              Hail             No. of Accidents  10199n   
5  (2016)  Northern Boarder     No. of Casualties - Dead    193n   
6  (2017)             Jazan             No. of Accidents  22161n   
7  (2017)           Al-Jouf             No. of Accidents  10469n   
8  (2017)             Total     No. of Casualties - Dead   7489n   
9  (2016)  Northern Boarder  No. of Casualties - Injured    631n   

                   geo_point_2d  
0   21.628962766, 41.2636898334  
1  23.2883307307, 50.1478302439  
2    27.90165499, 37.2598581394  
3   27.400845724, 41.4406960508  
4   27.400845724, 41.4406960508  
5   30.197074783, 41.8171186952  
6  17.2072313962, 42.6525807422  
7  29.9677445762, 39.5570371172  
8                           NaN  
9   30.197074783, 41.8171186952

In [4]:
accidents.tail()

Year    Region                    Indicator  Value  \
79  (2017)      Asir  No. of Casualties - Injured  2368n   
80  (2017)      Hail  No. of Casualties - Injured   773n   
81  (2017)  Al-Baaha     No. of Casualties - Dead   134n   
82  (2017)  Al-Baaha             No. of Accidents  1414n   
83  (2017)  Al-Baaha  No. of Casualties - Injured  1007n   

                    geo_point_2d  
79   19.1387847401, 42.941470036  
80   27.400845724, 41.4406960508  
81  20.1484058503, 41.4665484892  
82  20.1484058503, 41.4665484892  
83  20.1484058503, 41.4665484892

In [5]:
accidents.shape

(84, 5)

In [6]:
licenses.head(10)

Year Administritive Area Driving Liceses                 geo_point_2d
0  (2012)              Makkah         143954n  21.628962766, 41.2636898334
1  (1994)              Makkah          75349n  21.628962766, 41.2636898334
2  (1998)              Makkah          86082n  21.628962766, 41.2636898334
3  (2004)              Makkah          86636n  21.628962766, 41.2636898334
4  (2005)              Makkah          65099n  21.628962766, 41.2636898334
5  (2008)              Makkah          77013n  21.628962766, 41.2636898334
6  (2014)             Madinah          31447n  24.927534539, 39.4918237173
7  (2013)             Madinah          28513n  24.927534539, 39.4918237173
8  (2012)             Madinah          14459n  24.927534539, 39.4918237173
9  (1994)             Madinah           3767n  24.927534539, 39.4918237173

In [7]:
licenses.tail()

Year Administritive Area Driving Liceses                 geo_point_2d
345  (2000)              Tabouk           8473n   27.90165499, 37.2598581394
346  (2006)              Tabouk           8259n   27.90165499, 37.2598581394
347  (2013)                Hail           9731n  27.400845724, 41.4406960508
348  (1997)                Hail           6606n  27.400845724, 41.4406960508
349  (1999)                Hail           3139n  27.400845724, 41.4406960508

In [8]:
licenses.shape

(350, 4)

#### 3. Briefly describe the data

Take your time looking through the data and briefly describe the data in the markdown cell below. Note things about what the columns might mean, and the general information that is conveyed in the dataframe.

Answer:
- The accidents dataframe consists of yearly data (2016-2017) about vehicle accidents for regions in KSA, with various variables and values presented in a long format, also has a geo point column
- The licenses dataframe consists of yearly data over 25 years about the number of licenses issued per region in KSA, also has a geo point column as in accidents

#### 4a. How complete is the data?

Investigate missing values etc.

In [9]:
accidents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84 entries, 0 to 83
Data columns (total 5 columns):
Year            84 non-null object
Region          84 non-null object
Indicator       84 non-null object
Value           84 non-null object
geo_point_2d    78 non-null object
dtypes: object(5)
memory usage: 3.4+ KB


In [10]:
licenses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 350 entries, 0 to 349
Data columns (total 4 columns):
Year                   350 non-null object
Administritive Area    350 non-null object
Driving Liceses        350 non-null object
geo_point_2d           325 non-null object
dtypes: object(4)
memory usage: 11.0+ KB


### The geopoint column seems to be missing some entries in both accidents & licenses. All other observations seem to be nun-null with 84 & 350 entries respectively for other columns in accidents and licenses

#### 4b. Are there any obvious issues with the observations?

**What is the minimum *possible* value for each dataset? What is the maximum *possible* value?**

In [11]:
print(accidents.min())
print(accidents.max())

Year                   (2016)
Region               Al-Baaha
Indicator    No. of Accidents
Value                   1007n
dtype: object
Year                              (2017)
Region                             Total
Indicator    No. of Casualties - Injured
Value                             99925n
dtype: object


In [12]:
print(licenses.min())
print(licenses.max())

Year                    (1993)
Administritive Area    Al-Baha
Driving Liceses        100291n
dtype: object
Year                   (2017)
Administritive Area     Total
Driving Liceses         9983n
dtype: object


Data cannot be properly interpreted because of data type mismatch

In [13]:
accidents.describe()

Year  Region                    Indicator  Value  \
count       84      84                           84     84   
unique       2      14                            3     84   
top     (2017)  Makkah  No. of Casualties - Injured  2541n   
freq        42       6                           28      1   

                      geo_point_2d  
count                           78  
unique                          13  
top     27.90165499, 37.2598581394  
freq                             6

In [14]:
# check year range for dataset
accidents.Year.value_counts()

(2017)    42
(2016)    42
Name: Year, dtype: int64

In [15]:
licenses.describe()

Year Administritive Area Driving Liceses                geo_point_2d
count      350                 350             350                         325
unique      25                  14             350                          13
top     (2005)               Jazan          24515n  27.90165499, 37.2598581394
freq        14                  25               1                          25

In [16]:
# check year range for dataset
licenses.Year.value_counts().sort_index()

(1993)    14
(1994)    14
(1995)    14
(1996)    14
(1997)    14
(1998)    14
(1999)    14
(2000)    14
(2001)    14
(2002)    14
(2003)    14
(2004)    14
(2005)    14
(2006)    14
(2007)    14
(2008)    14
(2009)    14
(2010)    14
(2011)    14
(2012)    14
(2013)    14
(2014)    14
(2015)    14
(2016)    14
(2017)    14
Name: Year, dtype: int64

#### 5. What are your data types? 
Display the data types of each feature. 

In [17]:
# check data types with .dtypes()
accidents.dtypes

Year            object
Region          object
Indicator       object
Value           object
geo_point_2d    object
dtype: object

In [18]:
licenses.dtypes

Year                   object
Administritive Area    object
Driving Liceses        object
geo_point_2d           object
dtype: object

What did you learn?
- Do any of them seem odd?  
- Which ones are not as they should be?  

Answer:

- all columns for both datasets are of dtype object which usually means they are in string form
- Year could be numeric or a datetime, Region/Area a category, and driving 'liceses' & most values as numeric data after correcting them

#### 6. Fix incorrect data types
Based on what you discovered above, use appropriate methods to re-type incorrectly typed data.
- Define a function that will allow you to convert numerical columns to an appropriate numeric type. Use `map` or `apply` to change these columns in each dataframe.

In [19]:
def col_to_numeric(x):
    try:
        num_x = int(x)
        return num_x
    except:
        print('UNSUCCESFUL, CANNOT BE CONVERTED TO FLOAT')

- Make new columns `x` and `y` using `geo_point_2d`

In [20]:
# get x and y geo columns
accidents['x'] = accidents['geo_point_2d'].str.replace(' ', '').apply(lambda x: str(x)[:str(x).find(',')])
accidents['y'] = accidents['geo_point_2d'].str.replace(' ', '').apply(lambda y: str(y)[str(y).find(',')+1:])
accidents.head()

Year          Region                    Indicator   Value  \
0  (2016)          Makkah  No. of Casualties - Injured  12383n   
1  (2016)  Eastern Region     No. of Casualties - Dead   1113n   
2  (2016)          Tabouk     No. of Casualties - Dead    434n   
3  (2016)            Hail     No. of Casualties - Dead    299n   
4  (2016)            Hail             No. of Accidents  10199n   

                   geo_point_2d              x              y  
0   21.628962766, 41.2636898334   21.628962766  41.2636898334  
1  23.2883307307, 50.1478302439  23.2883307307  50.1478302439  
2    27.90165499, 37.2598581394    27.90165499  37.2598581394  
3   27.400845724, 41.4406960508   27.400845724  41.4406960508  
4   27.400845724, 41.4406960508   27.400845724  41.4406960508

- Fix any individual values preventing other columns from being the appropriate type.

In [21]:
# fix value issues including typos
accidents['Year'] = accidents['Year'].str.replace('(', '').str.replace(')', '')
accidents['Value'] = accidents['Value'].str.replace('n', '')
accidents['Region'] = accidents['Region'].str.replace('Boarder', 'Border')
accidents['Region'] = accidents['Region'].str.replace('Assir', 'Asir')
accidents.head()

Year          Region                    Indicator  Value  \
0  2016          Makkah  No. of Casualties - Injured  12383   
1  2016  Eastern Region     No. of Casualties - Dead   1113   
2  2016          Tabouk     No. of Casualties - Dead    434   
3  2016            Hail     No. of Casualties - Dead    299   
4  2016            Hail             No. of Accidents  10199   

                   geo_point_2d              x              y  
0   21.628962766, 41.2636898334   21.628962766  41.2636898334  
1  23.2883307307, 50.1478302439  23.2883307307  50.1478302439  
2    27.90165499, 37.2598581394    27.90165499  37.2598581394  
3   27.400845724, 41.4406960508   27.400845724  41.4406960508  
4   27.400845724, 41.4406960508   27.400845724  41.4406960508

In [22]:
licenses['Year'] = licenses['Year'].str.replace('(', '').str.replace(')', '')
licenses['Driving Liceses'] = licenses['Driving Liceses'].str.replace('n', '')
licenses['Administritive Area'] = licenses['Administritive Area'].str.replace('Boarder', 'Border')
licenses['Administritive Area'] = licenses['Administritive Area'].str.replace('Assir', 'Asir')
licenses.head()

Year Administritive Area Driving Liceses                 geo_point_2d
0  2012              Makkah          143954  21.628962766, 41.2636898334
1  1994              Makkah           75349  21.628962766, 41.2636898334
2  1998              Makkah           86082  21.628962766, 41.2636898334
3  2004              Makkah           86636  21.628962766, 41.2636898334
4  2005              Makkah           65099  21.628962766, 41.2636898334

- Finish your data modifications by making sure the columns are now typed appropriately.

In [23]:
# convert to numeric
accidents['Year'] = accidents.Year.apply(col_to_numeric)
accidents['Value'] = accidents.Value.apply(col_to_numeric)
accidents['Region'] = accidents.Region.astype('category')
licenses['Year'] = licenses.Year.apply(col_to_numeric)
licenses['Driving Liceses'] = licenses['Driving Liceses'].apply(col_to_numeric)
licenses['Administritive Area'] = licenses['Administritive Area'].astype('category')
accidents.head()

Year          Region                    Indicator  Value  \
0  2016          Makkah  No. of Casualties - Injured  12383   
1  2016  Eastern Region     No. of Casualties - Dead   1113   
2  2016          Tabouk     No. of Casualties - Dead    434   
3  2016            Hail     No. of Casualties - Dead    299   
4  2016            Hail             No. of Accidents  10199   

                   geo_point_2d              x              y  
0   21.628962766, 41.2636898334   21.628962766  41.2636898334  
1  23.2883307307, 50.1478302439  23.2883307307  50.1478302439  
2    27.90165499, 37.2598581394    27.90165499  37.2598581394  
3   27.400845724, 41.4406960508   27.400845724  41.4406960508  
4   27.400845724, 41.4406960508   27.400845724  41.4406960508

In [24]:
licenses.head()

Year Administritive Area  Driving Liceses                 geo_point_2d
0  2012              Makkah           143954  21.628962766, 41.2636898334
1  1994              Makkah            75349  21.628962766, 41.2636898334
2  1998              Makkah            86082  21.628962766, 41.2636898334
3  2004              Makkah            86636  21.628962766, 41.2636898334
4  2005              Makkah            65099  21.628962766, 41.2636898334

- Display the data types again to confirm they are correct.

In [25]:
accidents.dtypes

Year               int64
Region          category
Indicator         object
Value              int64
geo_point_2d      object
x                 object
y                 object
dtype: object

In [26]:
licenses.dtypes

Year                      int64
Administritive Area    category
Driving Liceses           int64
geo_point_2d             object
dtype: object

#### 7. Rename columns
Change the names of the columns to more expressive names so that you can tell the difference the Driving Licenses  columns and the & Traffic Accidents columns. Your solution should map all column names being changed at once (no repeated singular name-changes). **We will be combining these two datasets, and so you should name columns in an appropriate way**.

**Guidelines**:
- Column names should be all lowercase (you will thank yourself when you start pushing data to SQL later in the course)
- Column names should not contain spaces (underscores will suffice--this allows for using the `df.column_name` method to access columns in addition to `df['column_name']`.
- Column names should be unique and informative (the only feature that we actually share between dataframes is the state).
- Please also fix any typos you see in the column names.

In [27]:
#code

#### 8. Create a data dictionary

Now that we've fixed our data, and given it appropriate names, let's create a [data dictionary](http://library.ucmerced.edu/node/10249). 

A data dictionary provides a quick overview of features/variables/columns, alongside data types and descriptions. The more descriptive you can be, the more useful this document is.

Example of a Fictional Data Dictionary Entry: 

|Feature|Type|Dataset|Description|
|---|---|---|---|
|**county_pop**|*integer*|2010 census|The population of the county (units in thousands, where 2.5 represents 2500 people).| 
|**per_poverty**|*float*|2010 census|The percent of the county over the age of 18 living below the 200% of official US poverty rate (units percent to two decimal places 98.10 means 98.1%)|

[Here's a quick link to a short guide for formatting markdown in Jupyter notebooks](https://jupyter-notebook.readthedocs.io/en/stable/examples/Notebook/Working%20With%20Markdown%20Cells.html).

Provided is the skeleton for formatting a markdown table, with columns headers that will help you create a data dictionary to quickly summarize your data, as well as some examples. **This would be a great thing to copy and paste into your custom README for this project.**

|Feature|Type|Dataset|Description|
|---|---|---|---|
|column name|int/float/object|Driving_Licenses/Traffic_Accidents|This is an example| 


#### 9. Drop unnecessary rows

This study concerns accident and license in regions/area. Please do the followings:
0. Which rows are not associated with regions?
1. Confirm the numbers of all regions add up to total.
2. Remove the rows that are not associated with regions

In [28]:
#code

#### 10a. Data formats
Is the License data in long or wide format? How about the Accident data? And why?

In [29]:
#code

#### 10b. Changing formats
Which format would work best for both dataframes? Transform the datasets to the same format before merging in the next task.

In [30]:
#code

#### 11. Merge dataframes

Join the Driving Licenses & Traffic Accidents dataframes using the **region** and **year** in each dataframe as the key. Assign this to a new variable. Which join should you use to preserve the most data?

In [31]:
#code

#### 12. Save your cleaned, merged dataframe

Use a relative path to save out your data as `combined_datasets.csv`.

In [32]:
#code

## Exploratory Data Analysis


### Summary Statistics
Transpose the output of pandas `describe` method to create a quick overview of each numeric feature.

In [33]:
#Code:

#### Manually calculate standard deviation

$$\sigma = \sqrt{\frac{1}{n}\sum_{i=1}^n(x_i - \mu)^2}$$

- Write a function to calculate standard deviation using the formula above

In [34]:
#code

- Use a **dictionary comprehension** to apply your standard deviation function to each numeric column in the dataframe.  **No loops**  
- Assign the output to variable `sd` as a dictionary where: 
    - Each column name is now a key 
    - That standard deviation of the column is the value 
     
*Example Output :* `{'Driving_Licenses_x': 120, 'Traffic_Accidents_x': 120, ...}`

In [35]:
#Code:

Do your manually calculated standard deviations match up with the output from pandas `describe`? What about numpy's `std` method?

Answer:

#### Investigate trends in the data
Using sorting and/or masking (along with the `.head` method to not print our entire dataframe), consider the following questions:

- Which regions have the highest and lowest Driving Licenses based on years (1993 - 2017)
- Which regions have the highest and lowest mean Traffic Accidents numbers for years (2016 - 2017)
- What are the regions that have more Driving Licenses issued in each year than the year average?
- What are the regions that have more Traffic Accidents happened in each year than the year average?

Based on what you've just observed, have you identified any regions that you're especially interested in? **Make a note of these and state *why* you think they're interesting**.

**You should comment on your findings at each step in a markdown cell below your code block**. Make sure you include at least one example of sorting your dataframe by a column, and one example of using boolean filtering (i.e., masking) to select a subset of the dataframe.

In [36]:
#code

## Visualize the data

There's not a magic bullet recommendation for the right number of plots to understand a given dataset, but visualizing your data is *always* a good idea. Not only does it allow you to quickly convey your findings (even if you have a non-technical audience), it will often reveal trends in your data that escaped you when you were looking only at numbers.

Some recommendations on plotting:
- Plots have titles
- Plots have axis labels
- Plots have appropriate tick labels
- All text is legible in a plot
- Plots demonstrate meaningful and valid relationships
- Plots are interpreted to aid understanding

There is such a thing as too many plots, and there are a *lot* of bad plots. You might make some! (But hopefully not with the guided prompts below).

#### Use Seaborn's heatmap with pandas `.corr()` to visualize correlations between all numeric features

Heatmaps are generally not appropriate for presentations, and should often be excluded from reports as they can be visually overwhelming. **However**, they can be extremely useful in identify relationships of potential interest (as well as identifying potential collinearity before modeling).

*example*:
```python
sns.heatmap(df.corr())
```

Please take time to format your output, adding a title. Look through some of the additional arguments and options. (Axis labels aren't really necessary, as long as the title is informative).

#### Define a custom function to subplot histograms

We should write a function that will take the names of 2+ columns and subplot histograms. While you can use pandas plotting or Seaborn here, matplotlib gives you greater control over all aspects of your plots.

[Helpful Link for Plotting Multiple Figures](https://matplotlib.org/users/pyplot_tutorial.html#working-with-multiple-figures-and-axes)

Here's some starter code:

In [37]:
def subplot_histograms(dataframe, list_of_columns, list_of_titles, list_of_xlabels):
    nrows = int(np.ceil(len(list_of_columns)/2) # Makes sure you have enough rows
    fig, ax = plt.subplots(nrows=nrows, ncols=2) # You'll want to specify your figsize
    ax = ax.ravel() # Ravel turns a matrix into a vector, which is easier to iterate
    for i, column in enumerate(list_of_columns): # Gives us an index value to get into all our lists
        ax[i].hist(dataframe[column]) # feel free to add more settings
        # Set titles, labels, etc here for each subplot

SyntaxError: invalid syntax (<ipython-input-37-f8881e15e6cc>, line 3)

#### Plot and interpret histograms 
For each of the following:
- Driving Licenses number
- Traffic Accidents number

Please plot two additional histograms that will help with your understanding of the data

In [ ]:
# Code

#### Plot and interpret scatter plots

For each of the following:
- Driving Licenses vs. Traffic Accidents for 2017
- Driving Licenses vs. Traffic Accidents for 2016
- Driving Licenses for 2016 vs 2017
- Traffic Accidents for 2016 vs 2017

Plot the two variables against each other using matplotlib or Seaborn

Your plots should show:
- Two clearly labeled axes
- A proper title
- Using colors and symbols that are clear and unmistakable

**Feel free to write a custom function, and subplot if you'd like.** Functions save both time and space.


In [ ]:
# Code

#### Plot and interpret boxplots

For each numeric variable in the dataframe create a boxplot using Seaborn. Boxplots demonstrate central tendency and spread in variables. In a certain sense, these are somewhat redundant with histograms, but you may be better able to identify clear outliers or differences in IQR, etc.

Multiple values can be plotted to a single boxplot as long as they are of the same relative scale (meaning they have similar min/max values).

Each boxplot should:
- Only include variables of a similar scale
- Have clear labels for each variable
- Have appropriate titles and labels

In [ ]:
# Code

#### Feel free to do additional plots below
*(do research and choose your own chart types & variables)*

Are there any additional trends or relationships you haven't explored? Was there something interesting you saw that you'd like to dive further into? It's likely that there are a few more plots you might want to generate to support your narrative and recommendations that you are building toward. **As always, make sure you're interpreting your plots as you go**.

## Descriptive and Inferential Statistics

#### Summarizing Distributions

Above, we used pandas `describe` to provide quick summary statistics of our numeric columns. We also demonstrated many visual relationships.

As data scientists, having a complete understanding of data is imperative prior to modeling.

While we will continue to build our analytic tools, we know that measures of *central tendency*, *spread*, and *shape/skewness* provide a quick summary of distributions.

For each variable in your data, summarize the underlying distributions (in words & statistics)
 - Be thorough in your verbal description of these distributions.
 - Be sure to back up these summaries with statistics.

Answers:

#### We generally assuming that data we sample from a population will be normally distributed. Do we observe this trend?

Answer:

Does This Assumption Hold for:
    - Driving Licenses
    - Traffic Accidents
Explain your answers for each distribution and how you think this will affect estimates made from these data.

Answer:

#### Statistical Evaluation of Distributions 

**If you feel it's appropriate**, using methods we discussed in class, run hypothesis tests to compare variables of interest in our dataset. 

In [ ]:
# Code

## Outside Research

Based upon your observations, choose **three** regions that demonstrate interesting trends in the number of driving licenses and traffic accidents. Spend some time doing outside research on provincial and central policies that might influence these rates, and summarize your findings below. **Feel free to go back and create new plots that highlight these states of interest**. If you bring in any outside tables or charts, make sure you are explicit about having borrowed them. If you quote any text, make sure that it renders as being quoted. (Make sure that you cite your sources -- check with you local instructor for citation preferences).

Answer:

## Conclusions and Recommendations

- Based on your exploration of the data, what are you key takeaways and recommendations?
- Are there additional data you desire that would better inform your investigations?

Answer: